src/map/Map.js

```diff
@@ -520,43 +521,52 @@ L.Map = L.Evented.extend({
 
 	// private methods that modify map state
 
-	_resetView: function (center, zoom, preserveMapOffset, afterZoomAnim) {
+	_resetView: function (center, zoom) {
+		L.DomUtil.setPosition(this._mapPane, new L.Point(0, 0));
 
-		var zoomChanged = (this._zoom !== zoom);
+		var loading = !this._loaded;
+		this._loaded = true;
 
-		if (!afterZoomAnim) {
-			this.fire('movestart');
+		var zoomChanged = this._zoom !== zoom;
+		this
+			._moveStart(zoomChanged)
+			._move(center, zoom)
+			._moveEnd(zoomChanged);
 
-			if (zoomChanged) {
-				this.fire('zoomstart');
-			}
+		if (loading) {
+			this.fire('load');
 		}
+	},
 
-		this._zoom = zoom;
-		this._initialCenter = center;
-
-		if (!preserveMapOffset) {
-			L.DomUtil.setPosition(this._mapPane, new L.Point(0, 0));
+	_moveStart: function (zoomChanged) {
+		if (zoomChanged) {
+			this.fire('zoomstart');
 		}
+		return this.fire('movestart');
+	},
 
-		this._pixelOrigin = this._getNewPixelOrigin(center);
+	_move: function (center, zoom, data) {
+		if (zoom === undefined) {
+			zoom = this._zoom;
+		}
 
-		var loading = !this._loaded;
-		this._loaded = true;
+		var zoomChanged = this._zoom !== zoom;
 
-		this.fire('viewreset', {hard: !preserveMapOffset});

```

Comment:
We could add it back, although the semantics would change, since now changing zoom doesn't always imply recalculating everything (e.g. with fractions, or frames like flyTo does).